pipelinewise/cli/commands.py

```diff
@@ -64,17 +56,23 @@ def do_json_conf_validation(json_file: str, file_property: dict) -> bool:
     return False
 
 
-# pylint: disable=function-redefined)
-class TapParams(OriginalTapParams):
+@dataclass
+class TapParams:
     """
-    for overriding TapParam init and validating json properties
+    TapParams validates json properties.
     """
-    # pylint: disable=unused-argument
-    def __init__(self, *args, **kwargs):
-        super().__init__()
-        if not getattr(self, 'config', None):
+    id: str  # pylint: disable=invalid-name
```

Comment:
would it take much to rename `id` to something like `target_id`? Ideally we would avoid disabling pylint rules

Reply:
Yeah, I was torn on this. Might make it harder to get it merged upstream due to compatibility breakage? But I've gone with your suggestion.